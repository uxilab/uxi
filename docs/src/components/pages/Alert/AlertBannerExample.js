import React, {Â Component } from 'react';
import Alert from 'uxi/Alert';
import Button from 'uxi/Button';
import { P } from 'uxi/Classic';

class AlertBannerExample extends Component {
  constructor(props){
    super(props)
    this.state = {
      show1: false,
      show2: false,
    }
  }

  render () {
    return (
      <div>
        <Button onClick={() => this.setState({ show1: true }) } text="show A" />
        <br />
        <Button onClick={() => this.setState({ show2: true }) } text="show B" />

        {
          this.state.show1 && (
            <Alert isBanner >
              This is a banner Alert!
            </Alert>
          )
        }
        {
          this.state.show2 && (
            <Alert type="danger" isBanner>
              This is a banner Alert with too much text!
              Sunt reprehenderit incididunt occaecat laborum elit consequat reprehenderit.
              Velit nulla fugiat adipisicing proident aliqua culpa ex fugiat.
              Ipsum tempor nostrud amet officia consequat consectetur do labore.
              Incididunt culpa est irure nisi commodo adipisicing.
              Quis culpa ullamco est occaecat deserunt culpa occaecat.
              Officia elit qui ullamco do ipsum mollit aliquip sit excepteur.
              Elit deserunt magna pariatur deserunt culpa duis minim ex ad proident amet.
            </Alert>
          )
        }
    </div>
    )
  }
}

export default AlertBannerExample;
