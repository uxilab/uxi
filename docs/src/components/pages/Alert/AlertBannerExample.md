Alert takes a `isBanner` props.
