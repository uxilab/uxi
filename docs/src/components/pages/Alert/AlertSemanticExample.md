Alert take a `type` props allowing for smeantic coloring.

Alert take a `showClose` props whichj when true, makes the Alert "auto-desctructible" (tha alert will become invisible but will stil be mounted in the DOM)
