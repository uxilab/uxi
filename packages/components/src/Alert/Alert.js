import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
  Close,
  Issue,
  Alert as AlertIcon,
  Done,
} from '../Icons';
import AlertStyle from './Alert.style';

const AlertUI = styled.div`
  div{
    a, a:hover {
      color: white;
      text-decoration: underline;
    }
  }
`;

class Alert extends Component {
  static propTypes = {
    children: PropTypes.node,
    showClose: PropTypes.bool, // weird ?
    hideClose: PropTypes.bool, // weird ?
    onClose: PropTypes.func,
    type: PropTypes.oneOf(['danger', 'error', 'success', 'warning', 'information']),
    noIcon: PropTypes.bool,
    className: PropTypes.string,
    isBanner: PropTypes.bool,
    iconSize: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
    style: PropTypes.object, // eslint-disable-line react/forbid-prop-types
  };

  static defaultProps = {
    type: 'information',
    showClose: undefined,
    hideClose: undefined,
    className: '',
    noIcon: false,
    isBanner: false,
    iconSize: 24, // uxi default, src of thruth => SvgIcon
    onClose: null,
    // in case some text is not evaluated, the comp might appear like it has no children (text node)
    children: null, // we don't require children, alert always has at least 50px of height
    style: {},
  };

  constructor(props) {
    super(props);

    this.state = {
      isOpen: true,
    };
    this.close = this.close.bind(this);
    this.isControlled = !!this.props.onClose;
  }

  /**
   * Close the Alert message when the property showClose is set to 'true'
   */
  close(event) {
    const { onClose } = this.props;
    if (onClose) { onClose(event, this.state.isOpen); }
    if (!this.isControlled) {
      this.setState({ isOpen: false });
    }
  }

  render() {
    const {
      type,
      // TODO decide on a single behaviour
      showClose, // if banner defaults to showing X close, and you don't wnat one, ok
      hideClose, // but this is getting weird, we should settle on a posit or negatv word,
      noIcon,
      className,
      isBanner,
      iconSize,
      onClose,
      style,
    } = this.props;
    const { isOpen } = this.state;

    let wrapperStyles = { ...AlertStyle.alert };

    if (isBanner) {
      wrapperStyles = { ...wrapperStyles, ...AlertStyle.bannerAlert };
    }

    let closeContent;
    let iconContent;
    let IconComp;
    let mainAlertStyle = { ...AlertStyle.alertContent };
    if (!hideClose && (showClose || onClose || isBanner)) {
      closeContent = (
        <button key="closeIcon" onClick={this.close} style={AlertStyle.closeWrapper}>
          <Close color="white" />
        </button>
      );
      mainAlertStyle = {
        ...mainAlertStyle,
        marginRight: '50px',
      };
    }

    if (type === 'danger' || type === 'error') {
      wrapperStyles = { ...wrapperStyles, ...AlertStyle.danger };
      IconComp = Issue;
    } else if (type === 'warning') {
      wrapperStyles = { ...wrapperStyles, ...AlertStyle.warning };
      IconComp = AlertIcon;
    } else if (type === 'success') {
      wrapperStyles = { ...wrapperStyles, ...AlertStyle.success };
      IconComp = Done;
    } else {
      wrapperStyles = { ...wrapperStyles, ...AlertStyle.info };
      IconComp = AlertIcon;
    }

    if (!noIcon) {
      iconContent = (
        <IconComp color="white" style={AlertStyle.icon} size={iconSize} />
      );
    }

    if (!isOpen) {
      wrapperStyles = { ...wrapperStyles, display: 'none' };
    }

    wrapperStyles = {
      ...wrapperStyles,
      ...style, // overwrite prop from consumer
    };

    return (
      <AlertUI>
        <div style={wrapperStyles} className={className}>
          {closeContent}
          {iconContent}
          <div style={mainAlertStyle}>
            {this.props.children}
          </div>
        </div>
      </AlertUI>
    );
  }
}

export default Alert;
