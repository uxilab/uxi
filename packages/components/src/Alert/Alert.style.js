export default {
  alert: {
    margin: 0,
    padding: 0,
    boxSizing: 'border-box',
    position: 'relative',
    display: 'flex',
    alignItems: 'center',
    minHeight: '50px',
  },
  bannerAlert: {
    position: 'fixed',
    zIndex: 250,
    top: 0,
    left: 0,
    right: 0,
    bottom: 'auto',
  },
  info: {
    color: '#fff',
    backgroundColor: 'rgb(49, 112, 143)',
    borderColor: 'rgb(49, 112, 143)',
  },
  danger: {
    backgroundColor: '#EF5858',
    color: '#fff',
  },
  warning: {
    backgroundColor: '#FF9800',
    color: '#fff',
  },
  success: {
    backgroundColor: '#009688',
    color: '#fff',
  },
  icon: {
    margin: '8px 8px 8px 16px',
    minWidth: '24px',
  },
  alertContent: {
    display: 'inline-block',
    boxSizing: 'border-box',
    padding: '15px',
    width: '100%',
  },
  closeWrapper: {
    position: 'absolute',
    cursor: 'pointer',
    right: '15px',
    top: '13px',
    padding: 0,
    margin: 0,
    border: 'none',
    background: 'transparent',
  },
};
