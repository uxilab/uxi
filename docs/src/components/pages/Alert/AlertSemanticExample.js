import React from 'react';
import Alert from '../../../../../src/Alert';
import { P } from '../../../../../src/Classic';

const AlertSemanticExample = () => (
  <div>
    <Alert type="error" noIcon showClose rounded>
      This is an danger Alert without icon and close btn! Daaaanger zone....
    </Alert>
    <br />
    <Alert type="danger" showClose rounded>
      This is an danger Alert with icon and close btn! Daaaanger zone....
    </Alert>
    <br />
    <Alert rounded>
      This is an default Alert with an icon!
    </Alert>
    <br />
    <Alert type="success">
      This is an success Alert with an icon!
    </Alert>
    <br />
    <Alert type="warning">
      This is an warn Alert with an icon!
    </Alert>
    <br />
    <Alert showClose>
      This is an Alert with an icon and close btn!
    </Alert>
    <br />
    <Alert noIcon showClose rounded>
      This is an Alert without icon but with a close btn!
    </Alert>
  </div>
);

export default AlertSemanticExample;
