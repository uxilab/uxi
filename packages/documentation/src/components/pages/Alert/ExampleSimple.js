import React from 'react';
import Alert from '../../../../../src/Alert';
import { P } from '../../../../../src/Classic';

const ExampleSimple = () => (
  <div>
    <Alert type="danger" noIcon style={{ width: '150px' }}>
      This is an danger Alert without icon and close btn!
      It has a fixed width of 150px
      Daaaanger zone....
    </Alert>
    <br />
    <Alert type="danger" noIcon style={{ maxWidth: '150px' }}>
      This is an danger Alert without icon and close btn!
      It has a fixed width (max-width) of 150px
      Daaaanger zone....
    </Alert>
    <br />
    <Alert type="danger" noIcon style={{ maxWidth: '150px' }}>
      I'm an alert that has a link in its children <a href="#">link</a>
    </Alert>
    <br />
    <Alert type="danger" noIcon >
      This is an danger Alert without icon and close btn! Daaaanger zone....
    </Alert>
    <br />
    <Alert type="default">
      This is an default Alert with an icon!
    </Alert>
    <br />
    <Alert type="success">
      This is an success Alert with an icon!
    </Alert>
    <br />
    <Alert type="warning">
      This is an warn Alert with an icon!
    </Alert>
    <br />
    <Alert>
      This is an Alert with an icon and close btn!
    </Alert>
    <br />
    <Alert noIcon showClose>
      This is an Alert without icon but with a close btn!
    </Alert>
    <br />
    <Alert>
      This is an monstruous Alert with too much text!
      Sunt reprehenderit incididunt occaecat laborum elit consequat reprehenderit.
      Velit nulla fugiat adipisicing proident aliqua culpa ex fugiat.
      Ipsum tempor nostrud amet officia consequat consectetur do labore.
      Incididunt culpa est irure nisi commodo adipisicing.
      Quis culpa ullamco est occaecat deserunt culpa occaecat.
      Officia elit qui ullamco do ipsum mollit aliquip sit excepteur.
      Elit deserunt magna pariatur deserunt culpa duis minim ex ad proident amet voluptate.
    </Alert>
    <br />
    <Alert showClose>
      This is an monstruous Alert with too much text!
      Sunt reprehenderit incididunt occaecat laborum elit consequat reprehenderit.
      Velit nulla fugiat adipisicing proident aliqua culpa ex fugiat.
      Ipsum tempor nostrud amet officia consequat consectetur do labore.
      Incididunt culpa est irure nisi commodo adipisicing.
      Quis culpa ullamco est occaecat deserunt culpa occaecat.
      Officia elit qui ullamco do ipsum mollit aliquip sit excepteur.
      Elit deserunt magna pariatur deserunt culpa duis minim ex ad proident amet.
    </Alert>
    <br />
    <Alert isBanner>
      This is a banner Alert!
    </Alert>
    <br />
    <Alert type="danger" isBanner>
      This is a banner Alert with too much text!
      Sunt reprehenderit incididunt occaecat laborum elit consequat reprehenderit.
      Velit nulla fugiat adipisicing proident aliqua culpa ex fugiat.
      Ipsum tempor nostrud amet officia consequat consectetur do labore.
      Incididunt culpa est irure nisi commodo adipisicing.
      Quis culpa ullamco est occaecat deserunt culpa occaecat.
      Officia elit qui ullamco do ipsum mollit aliquip sit excepteur.
      Elit deserunt magna pariatur deserunt culpa duis minim ex ad proident amet.
    </Alert>
    <P>Alert without content has a min height of 50px :</P>
    <Alert type="danger"/>

    <Alert type="danger" iconSize={64} >
      I'm an alert with a big icon (custom sized)
    </Alert>
    <h2> Alert with exit handler </h2>
    <Alert type="danger" onClose={() => { window.alert('exited Alert'); }}>
      I'm an alert with a big icon (custom sized)
    </Alert>
  </div>
);

export default ExampleSimple;
