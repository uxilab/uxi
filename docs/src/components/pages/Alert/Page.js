import React from 'react';
import { P } from 'uxi/Classic';
import { Title } from 'uxi/Text';
import CodeExample from '../../CodeExample';

import AlertSemanticExample from './AlertSemanticExample';
import RAWAlertSemanticExample from '!raw-loader!./AlertSemanticExample';
import RAWMDAlertSemanticExample from '!raw-loader!./AlertSemanticExample.md';

import AlertBannerExample from './AlertBannerExample';
import RAWAlertBannerExample from '!raw-loader!./AlertBannerExample';
import RAWMDAlertBannerExample from '!raw-loader!./AlertBannerExample.md';

const AlertPage = () => (
  <div>
    <Title text="Alerts" />
    <ul>
      <li>
        <CodeExample
          code={RAWAlertSemanticExample}
          component
          title="Semantic alerts"
          description={RAWMDAlertSemanticExample}
          hasPadding
        >
          <AlertSemanticExample />
        </CodeExample>
      </li>
      <li>
        <CodeExample
          code={RAWAlertBannerExample}
          component
          title="Semantic alerts"
          description={RAWMDAlertBannerExample}
          hasPadding
        >
          <AlertBannerExample />
        </CodeExample>
      </li>
    </ul>
  </div>
);

export default AlertPage;
