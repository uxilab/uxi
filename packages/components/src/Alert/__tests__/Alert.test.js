import React from 'react';
import Alert from '../Alert';

it('should render an Alert', () => {
  const child = 'This is a message';
  const wrapper = shallow(
    <Alert>{child}</Alert>,
  );
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.contains('This is a message')).toEqual(true);
});
