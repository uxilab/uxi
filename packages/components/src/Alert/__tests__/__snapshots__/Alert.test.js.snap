// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an Alert 1`] = `
<styled.div>
  <div
    className=""
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(49, 112, 143)",
        "borderColor": "rgb(49, 112, 143)",
        "boxSizing": "border-box",
        "color": "#fff",
        "display": "flex",
        "margin": 0,
        "minHeight": "50px",
        "padding": 0,
        "position": "relative",
      }
    }
  >
    <Alert
      color="white"
      size={24}
      style={
        Object {
          "margin": "8px 8px 8px 16px",
          "minWidth": "24px",
        }
      }
    />
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "padding": "15px",
          "width": "100%",
        }
      }
    >
      This is a message
    </div>
  </div>
</styled.div>
`;
